src/test/run-make-fulldeps/target-specs/foo.rs

```diff
@@ -11,7 +11,7 @@ trait Sized {}
 auto trait Freeze {}
 
 #[lang = "start"]
-fn start(_main: *const u8, _argc: isize, _argv: *const *const u8) -> isize {
+fn start<T>(_main: *const u8, _argc: isize, _argv: *const *const u8) -> isize {
```

Comment:
```suggestion
fn start<T>(_main: fn() -> T, _argc: isize, _argv: *const *const u8) -> isize {
```

also

```rust
#[lang = "start"]
fn start(_main: fn(), _argc: isize, _argv: *const *const u8) -> isize {
    0
}

fn main() {}
```

currently works and behaves as you would expect.

Reply:
While that isn't strictly needed because it's unused and raw ptrs and fn ptrs have the same ABI, it is more correct, thanks.